ddl/partition.go

```diff
@@ -1378,6 +1384,27 @@ func buildCheckSQLForRangeColumnsPartition(pi *model.PartitionInfo, index int, s
 	}
 }
 
+func buildCheckSQLForListPartition(pi *model.PartitionInfo, index int, schemaName, tableName model.CIStr) string {
+	inValues := getInValues(pi, index)
+	sql := fmt.Sprintf("select 1 from `%s`.`%s` where %s not in (%s) limit 1", schemaName.L, tableName.L, pi.Expr, inValues)
+	return sql
+}
+
+func buildCheckSQLForListColumnsPartition(pi *model.PartitionInfo, index int, schemaName, tableName model.CIStr) string {
+	colName := pi.Columns[0].L
+	inValues := getInValues(pi, index)
+	sql := fmt.Sprintf("select 1 from `%s`.`%s` where %s not in (%s) limit 1", schemaName.L, tableName.L, colName, inValues)
+	return sql
+}
+
+func getInValues(pi *model.PartitionInfo, index int) string {
+	inValues := make([]string, 0, len(pi.Definitions[index].InValues))
```

Comment:
why not `strings.Join(pi.Definitions[index].InValues, ",")` ?

Reply:
`pi.Definitions[index].InValues` is a `[][]string`

Reply:
got it

Reply:
Any other problems in this issue?